dashboard/src/pages/events/AllProposals.vue

```diff
@@ -0,0 +1,252 @@
+<template>
+  <Header />
+  <div
+    v-if="event.loading"
+    class="max-w-screen-lg mx-auto mt-8 flex items-center justify-center min-h-[320px]"
+  >
+    <LoadingText class="text-lg" text="Loading Proposals..." />
+  </div>
+  <div v-else class="container mx-auto px-4">
+    <!-- Back button -->
+    <Breadcrumb class="mt-2" :items="breadcrumb_items" />
+    <EventHeader
+      :event="event.data"
+      class="my-4 border-b border-gray-900 pb-4"
+    />
+
+    <div v-if="cfp.data && proposals.data && proposalLikes.data">
+      <AllProposalsBanner :event="event.data" :cfp="cfp.data" class="py-2" />
+
+      <!-- Search and filters -->
+      <div class="flex flex-col gap-y-3 justify-between my-7 md:flex-row">
+        <div class="md:w-1/2">
+          <input
+            v-model="searchQuery"
+            type="text"
+            placeholder="Search by proposal title"
+            class="w-full px-4 py-2 border rounded-sm bg-gray-300"
+          />
+        </div>
+        <div class="flex gap-x-2 items-center">
+          <div class="relative w-40">
+            <button
+              @click="toggleSessionTypeDropdown"
+              class="px-4 py-2 flex gap-x-1 justify-between items-center border rounded-sm bg-gray-800 text-white w-40"
+            >
+              <p>{{ selectedSessionType || 'Session Type' }}</p>
+              <ChevronDown />
+            </button>
+            <div
+              v-if="showSessionTypeDropdown"
+              class="absolute z-10 mt-1 w-full bg-white border rounded-sm shadow-lg"
+            >
+              <div
+                v-for="type in sessionTypes"
+                :key="type"
+                @click="selectSessionType(type)"
+                class="px-4 py-2 hover:bg-gray-100 cursor-pointer"
+              >
+                {{ type }}
+              </div>
```

Comment:
I've used a styled version of select drop-down here. Hence it had to be monitored via the given controls. - toggles on button clicks, states etc. A normal select gives a browser rendered drop-down.

<img width="462" alt="Screenshot 2024-09-03 at 10 26 33 PM" src="https://github.com/user-attachments/assets/42b0965f-7f6a-438d-9543-d3a7db15d9dd">

Frappe UI's select gives a browser rendered drop-down which cannot be styled as given. Hence didn't use it for this case. 
![Screenshot 2024-09-03 at 10 28 30 PM](https://github.com/user-attachments/assets/6ddc5747-69e5-4e4d-a4e0-ffab4f536661)


This was purely a style choice - can be changed if needed.

Reply:
https://github.com/fossunited/fossunited/pull/589/files/cb99fca591d632446d2fc4ebcbeb73a2b0cc31cf#r1742396123

Reply:
I think you can style select fields as required. I'm pretty sure I have done the same in one of the files I mentioned above. 

I don't know if style should be kept above functionality and quality of code 😬

I really don't feel this is the way to handle this. Feels unnecessarily complex, for something that can be easier.

Reply:
> I think you can style select fields as required. I'm pretty sure I have done the same in one of the files I mentioned above.

It's the drop down that I am mentioning about - you cannot style a browser rendered component consistently across to support all browsers - https://blog.logrocket.com/creating-custom-select-dropdown-css/#cross-browser-compatibility-responsiveness

And hence, as I mentioned it was custom built, which can also be used as a component in future for other drop-downs if needed.

But since you've mentioned that we don't need it, I will revert the custom component.

Reply:
On second thoughts, I'll try using the vue-select component here rather than the custom built one - https://vue-select.org/guide/css.html#dark-mode-example